From 13be28b439413351b2aacb7f72ba35e499959e05 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 27 Oct 2020 15:00:46 +0800
Subject: [PATCH 12/12] gsttools: videowidget: Avoid re-creating window during
 destroying

There's a WinIdChange event when clearing window's winId.

Use internalWinId() instead of winId() in WinIdChange event to avoid
re-creating window.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/gsttools/qgstreamervideowidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gsttools/qgstreamervideowidget.cpp b/src/gsttools/qgstreamervideowidget.cpp
index ea37f18..5f17aab 100644
--- a/src/gsttools/qgstreamervideowidget.cpp
+++ b/src/gsttools/qgstreamervideowidget.cpp
@@ -182,7 +182,7 @@ bool QGstreamerVideoWidgetControl::eventFilter(QObject *object, QEvent *e)
 {
     if (m_widget && object == m_widget) {
         if (e->type() == QEvent::ParentChange || e->type() == QEvent::Show || e->type() == QEvent::WinIdChange) {
-            WId newWId = m_widget->winId() ? m_widget->videoWinId() : 0;
+            WId newWId = m_widget->internalWinId() ? m_widget->videoWinId() : 0;
             if (newWId != m_windowId)
                 m_videoOverlay.setWindowHandle(m_windowId = newWId);
         }
-- 
2.20.1

